src/main/java/com/workday/meta/PackageTree.java

```diff
@@ -124,36 +122,34 @@ private Node findDeepestMatchingNode(List<PackageElement> hierarchy) {
      * rootNode}.
      * @param rootNode The Node to start after.
      */
-    private void addPackageToNode(List<PackageElement> hierarchy, Node rootNode) {
+    private void addPackageToNode(List<String> hierarchy, Node rootNode) {
         if (hierarchy.isEmpty()) {
             return;
         }
 
-        PackageElement root = hierarchy.get(0);
-        String rootName = root.getSimpleName().toString();
-        Node parentNode = rootNode.children.get(rootName);
+        String root = hierarchy.get(0);
+        Node parentNode = rootNode.children.get(root);
```

Comment:
I think these names don't make a lot of sense (even though they are the original names). How about we rename parentNode to nextNode, and we rename root (not rootNode) to nextPackagePoint?